src/analyser.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  type Atrule,
+  type CssNode,
+  type Declaration,
+  type DSNodeString,
+  find,
+  generate,
+  parse as parseCss,
+  walk,
+  type WalkContext,
+} from 'css-tree'
+import { difference, escapeComments, exists, getExtension } from './utils'
+import type { CssFileMeta, FontFaceMeta, ImportResolvers, InternalLogger, TargetAttributes } from './types'
+import { isAbsolute } from 'node:path'
+import { existsSync, readFileSync } from 'node:fs'
+import { NODE_TYPE } from './constants'
+
+export class Analyzer {
+  constructor (
+    private readonly resolvers: ImportResolvers,
+    private readonly logger: InternalLogger,
+  ) { }
+
+  private readonly getFontName = (node: Declaration): string => {
+    const value = node.value
+    if (value.type === 'Value') {
+      return (value.children.first as DSNodeString).value
+    }
+    return value.value
+  }
+
+  findAtRules = (ast: CssNode): TargetAttributes => {
+    const nodes = Object.fromEntries<CssNode[]>(
+      Object.values(NODE_TYPE).map(type => [type, []]),
+    ) as TargetAttributes
+
+    walk(ast, {
+      visit: 'Atrule',
+      enter (this: WalkContext, node: Atrule) {
+        if (node.name === NODE_TYPE.IMPORT || node.name === NODE_TYPE.FONT_FACE) {
```

Comment:
what about `@mixin` ?

Reply:
good point. Thank you for your attention. Reworked to another solution